test/phabricator_tests/test_phabricator.py

```diff
@@ -0,0 +1,658 @@
+                phab=fake_phab
+            )
+
+        fake_phab.differential.getrevisioncomments.assert_called_with(
+            ids=[0]
+        )
+
+    @patch(PATH + 'PhabricatorService.user_query_usernames')
+    def test_generate_phids_successful(self, mock_user_query_usernames):
+        """
+        Tests generate_phids function (successful, user_query_usernames does not fail)
+        """
+        expected_list_of_phids, expected_raw_response = self.mock.generate_phids_response()
+        mock_user_query_usernames.return_value = self.mock.user_query_usernames()
+
+        list_of_phids, raw_response = PhabricatorService().generate_phids(
+            "user_names",
+            None)
+
+        self.assertEqual(expected_list_of_phids, list_of_phids)
+        self.assertEqual(expected_raw_response, raw_response)
+        mock_user_query_usernames.assert_called_with(
+            "user_names",
+            None
+        )
+
+    @patch(PATH + 'PhabricatorService.user_query_usernames',
+           side_effect=InternetError)
+    def test_generate_phids_failure(self, mock_user_query_usernames):
+        """
+        Tests generate_phids function (successful, user_query_usernames does fail)
+        """
+
+        with self.assertRaises(InternetError):
+            list_of_phids, raw_response = PhabricatorService().generate_phids(
+                "user_names",
+                None)
+
+        mock_user_query_usernames.assert_called_with(
+            "user_names",
+            None
+        )
+
+    @patch(PATH + 'PhabricatorService.get_comments')
+    @patch(PATH + 'PhabricatorService.get_last_comment')
+    @patch(PATH + 'PhabricatorService.time_from_epoch')
+    @patch(PATH + 'PhabricatorService.check_request_state')
+    @patch(PATH + 'PhabricatorService.has_new_comments')
+    @patch(PATH + 'PhabricatorService.author_data')
+    @patch(PATH + 'PhabricatorReview')
+    def test_get_reviews_successful_no_last_comment(self,mock_PhabricatorReview,
+                                                    mock_author_data,
+                                                    mock_has_new_comments,
+                                                    mock_check_request_state,
+                                                    mock_time_from_epoch,
+                                                    mock_get_last_comment,
+                                                    mock_get_comments):
+        """
+        Tests get_reviews function with simple parameters (i.e. no show_last_comment, duration, value etc.)
+        """
+
+        mock_PhabricatorReview.return_value = MagicMock()
+        mock_time_from_epoch.return_value = "mock_date"
+        fake_phab = MagicMock()
+        fake_reviews = self.mock.get_reviews()
+        mock_get_comments.return_value = "test comment"
+        mock_check_request_state.return_value = True
+        mock_author_data.return_value = ({'userName': 'mock_user', 'image': 'mock.com'}, [])
```

Comment:
These lines are pretty repetitive. Maybe you can come up with something that would reduce it. For example you can create another class, TestPhabricatorGetReviews and put these definition lines to `setUp` or even `setUpClass` method.